command/genesis/params.go

```diff
@@ -187,6 +192,16 @@ func (p *genesisParams) validateFlags() error {
 		}
 	}
 
+	if err := types.IsValidAddress(params.stakeToken); err != nil {
+		return fmt.Errorf("stake token address is not a valid address: %w", err)
+	}
+
+	params.stakeTokenAddr = types.StringToAddress(params.stakeToken)
+
+	if params.stakeTokenAddr == types.ZeroAddress {
+		return errStakeTokenIsZeroAddress
+	}
```

Comment:
Consider moving this validation where we are checking if it is `polybft` consensus (`if p.isPolyBFTConsensus()`).